### PR TITLE
tests(smoke): add tests for report-assert.js

### PR DESCRIPTION
**Pull request by connorjclark**
Adding some tests here, will help with #14128

### CODE REVIEW

**Comment by brendankenny** [comment]
This is another instance of an old discussion, but the report assertions are simple and compose simply and so would lend themselves to regular unit tests very well: (1) Do these assertions work, then (2) Do these assertions compose.

Snapshot tests can be good for complicated data and especially for complicated data that's expected to change often, but they come at the price of being verbose and having a low signal/noise ratio. It's difficult to tell what a test is for (especially when it fails on something that it's ostensibly _not_ for), and there's a ton of input and a ton of output to wade through to figure that out.

The benefit of the assertions being written (relatively) generally is that an entire LHR isn't needed to test if `_includes` is working. It just needs needs the single entry, which also means it's simple (and short) to test multiple variations of it.

File: lighthouse-cli/test/smokehouse/report-assert-test.js [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+ */
+
+/* eslint-disable no-control-regex */
+
+import {findDifference, getAssertionReport} from './report-assert.js';
+import {readJson} from '../../../root.js';
+
+describe('findDifference', () => {
+  const testCases = {
+    'works (trivial passing)': {
```
Comment (connorjclark) [inline lighthouse-cli/test/smokehouse/report-assert-test.js:14]: @brendankenny how's this?

File: lighthouse-cli/test/smokehouse/report-assert-test.js [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+/**
+ * Removes ANSI codes.
+ * TODO: should make it so logger can disable these.
+ * @param {string} text
+ */
+function clean(text) {
+  return text
+    .replace(/\x1B.*?m/g, '')
+    .replace(/\x1b.*?m/g, '')
+    .replace(/[✘×]/g, 'X')
```
Comment (connorjclark) [inline lighthouse-cli/test/smokehouse/report-assert-test.js:162]: any idea why logger does different characters for windows???

File: lighthouse-cli/test/smokehouse/report-assert-test.js [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+      actual: {prices: [0, 1, 2, 3, 4, 5]},
+      expected: {prices: {_includes: [2], _excludes: [2, 1]}},
+      diff: {path: '.prices', actual: 1, expected: {
+        expectedExclusion: 1,
+        message: 'Expected to not find matching entry via _excludes',
+      }},
+    },
+  };
+
+  for (const [testName, {actual, expected, diff}] of Object.entries(testCases)) {
```
Comment (connorjclark) [inline lighthouse-cli/test/smokehouse/report-assert-test.js:146]: @exterkamp i basically wrote a go test

**Pull request merged**